profiling/src/profiling.rs

```diff
@@ -848,84 +793,40 @@ impl Profiler {
-                            value: LabelValue::Str(
-                                format!("{}", context.local_root_span_id).into(),
-                            ),
-                        });
-                        labels.push(Label {
-                            key: "span id".into(),
-                            value: LabelValue::Str(format!("{}", context.span_id).into()),
-                        });
-                    }
-                }
-
-                let mut tags = locals.tags.clone();
-
-                if let Some(version) = PHP_VERSION.get() {
-                    /* This should probably be "language_version", but this is
-                     * the tag that was standardized for this purpose. */
-                    let tag = Tag::new("runtime_version", version)
-                        .expect("runtime_version to be a valid tag");
-                    tags.push(tag);
-                }
-
-                if let Some(sapi) = crate::SAPI.get() {
-                    match Tag::new("php.sapi", sapi.to_string()) {
-                        Ok(tag) => tags.push(tag),
-                        Err(err) => warn!("Tag error: {}", err),
-                    }
-                }
-
-                let message = SampleMessage {
-                    key: ProfileIndex {
-                        sample_types,
-                        tags,
-                        endpoint: locals.uri.clone(),
-                    },
-                    value: SampleData {
-                        frames,
-                        labels: labels.clone(),
-                        sample_values,
-                    },
-                };
-
-                match self.send_sample(message) {
+                let labels = self.message_labels();
+
+                match self.send_sample(self.prepare_sample_message(
+                    frames,
+                    0,
+                    0,
+                    None,
+                    count,
+                    bytes,
+                    &labels,
+                    locals
+                )) {
                     Ok(_) => trace!(
-                        "Sent stack sample of depth {} with labels {:?} to profiler.",
+                        "Sent stack sample of depth {} with size {}, labels {:?} and count {} to profiler.",
                         depth,
-                        labels
+                        bytes,
+                        labels,
+                        count,
                     ),
                     Err(err) => warn!(
-                        "Failed to send stack sample of depth {} with labels {:?} to profiler: {}",
-                        depth, labels, err
+                        "Failed to send stack sample of depth {} with size {}, labels {:?} and count {} to profiler: {}",
+                        depth,
+                        bytes,
+                        labels,
+                        count,
+                        err
```

Comment:
There are a few places where we can do things like this:

```suggestion
                        "Failed to send stack sample of depth {depth} with size {bytes}, labels {labels:?} and count {count} to profiler: {err}"
```

Landed in Rust v1.58.

Reply:
Thats nice, especially if there are more arguments. I updated also others that I could find